### ISSUE TITLE
How to make it work with laravel/mix

### ISSUE THREAD

**Issue by Andrew-web-coder** [microsoft/monaco-editor-webpack-plugin#143]
I made a fresh page with the latest laravel and added this to mix config:

```JS
const MonacoWebpackPlugin = require("monaco-editor-webpack-plugin");

mix.webpackConfig({
    plugins: [
        new MonacoWebpackPlugin({
            languages: ["html", "css", "javascript"],
        }),
    ],
});
```

I see no errors, but it does not seem to work at all, webpack creates a ton of files for each language.

**Comment by chaoyenpo** [comment]
This has nothing to do with worker.

```js
import * as monaco from 'monaco-editor'

```
change to
```js
import * as monaco from 'monaco-editor/esm/vs/editor/editor.api';
```

There are instructions in the README.

**Comment by philipriley** [comment]
I'm not sure whether to create a separate issue as it may be related to this one. 
Instead of laravel mix, I am using Ruby on Rails with Webpacker, but I guess it's similar. Both are wrappers around webpack. 
I can actually get the editor to render and it seems the plugin is doing it's job, but my bundle size is enormous, even though I'm excluding a lot.
I'm importing the editor as such:
`import * as monaco from 'monaco-editor/esm/vs/editor/editor.api';`
and my webpack config looks like this (it gets merged into the main webpack config later):

``` javascript
const MonacoWebpackPlugin = require('monaco-editor-webpack-plugin');

module.exports = {	
  plugins: [
    new MonacoWebpackPlugin({
      languages: ['yaml'],
      features: [
        '!accessibilityHelp',
        '!bracketMatching',
        '!caretOperations',
        '!clipboard',
        '!codeAction',
        '!codelens',
        '!colorDetector',
        '!comment',
        '!contextmenu',
        '!coreCommands',
        '!cursorUndo',
        '!dnd',
        '!folding',
        '!fontZoom',
        '!format',
        '!gotoError',
        '!gotoLine',
        '!gotoSymbol',
        '!hover',
        '!iPadShowKeyboard',
        '!inPlaceReplace',
        '!inspectTokens',
        '!linesOperations',
        '!links',
        '!multicursor',
        '!parameterHints',
        '!quickCommand',
        '!quickOutline',
        '!referenceSearch',
        '!rename',
        '!smartSelect',
        '!snippets',
        '!suggest',
        '!toggleHighContrast',
        '!toggleTabFocusMode',
        '!transpose',
        '!wordHighlighter',
        '!wordOperations',
        '!wordPartOperations'
			],
      filename: "[name].[contenthash].bundle.js"
		}),
  ],
  devtool: 'source-map',
};
```
My bundle size is around 70mb no matter what I do. The editor is usable, though when using the search field, the icons for 'find', 'replace' etc are garbled. Not sure if this is related in any way. 
![Screenshot from 2021-06-29 22-29-29](https://user-images.githubusercontent.com/1864400/123863286-ae888080-d929-11eb-8c88-7a47d90b89b3.png)

Can anyone spot something I am doing wrong?
Using monaco-editor `v.0.25.2` and monaco-editor-webpack-plugin `v4.0.0`.

**Comment by alexdima** [comment]
@philipriley That is quite large. Maybe you could check that sourcemaps are not inlined or something and maybe you could use a minifier? You can try this sample https://github.com/microsoft/monaco-editor-samples/tree/main/browser-esm-webpack-monaco-plugin to get a feel for the sizes you could expect.

**Comment by philipriley** [comment]
Thanks for replying so quick @alexdima. Yes, webpacker was not minifying in production and was also inlining the source maps (which I believe is by default with webpacker). So gzipped and minified, the main application bundle is now 1.6mb, which is much better. I could probably improve on this by splitting monaco editor out into its own chunk, but this will do fine for now. Thanks for the advice!

**Comment by alexdima** [comment]
> just a last question regarding the css issue I am having with the search bar. Can you think of what this could be?

🤷‍♂️ . All our icons are done using a font that we ship. Webpack needs to always be told to transport that (e.g. [here](https://github.com/microsoft/monaco-editor-webpack-plugin/blob/b3114eb9247421c3ff2ecd8713b1b4e57dcabefa/test/webpack.config.js#L17)). I would investigate if the font is generated in the bundles folder, if the font is loaded, and finally if there is some Unicode mixture with the class names of those icons (those icons work by using special unicode points that are reserved).

**Comment by philipriley** [comment]
Yes I have a loader setup to handle ttf files. It looks like it is more of an issue on my end as the font is generated in the bundles folder as expected. Thanks for all your help!

**Comment by Cannonb4ll** [comment]
@philipriley Did you ever manage to get around this? I'm currently experiencing the same with the codicon.ttf not being able to be loaded into my project.

**Comment by philipriley** [comment]
@Cannonb4ll unfortunately not. I have just not had enough time to focus on it. I raised an issue over in the [Rails Webpacker repo](https://github.com/rails/webpacker/issues/3077?notification_referrer_id=MDE4Ok5vdGlmaWNhdGlvblRocmVhZDIxMjY5MjI2NTA6MTg2NDQwMA%3D%3D#issuecomment-900937229), but have not had much luck there either. If you do ever figure it out, I'd love to hear how you did it!

**Comment by Cannonb4ll** [comment]
@philipriley Yeah so what I did yesterday what worked is add this inside my scss files:

```css
@font-face {
  font-family: 'codicon';
  src: url('../fonts/monaco/codicon.ttf') format('truetype');
}
```

Make sure to change the url to match your projects public dir where the .ttf file is. This worked for me temporary.
